src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Management.Automation.Internal;
+using System.Management.Automation.Runspaces;
+using System.Management.Automation.Subsystem.Prediction;
+using System.Threading;
+
+namespace System.Management.Automation.Subsystem.Feedback
+{
+    /// <summary>
+    /// Interface for implementing a feedback provider on command failures.
+    /// </summary>
+    public interface IFeedbackProvider : ISubsystem
+    {
+        /// <summary>
+        /// Default implementation. No function is required for a feedback provider.
+        /// </summary>
+        Dictionary<string, string>? ISubsystem.FunctionsToDefine => null;
+
+        /// <summary>
+        /// Default implementation for `ISubsystem.Kind`.
+        /// </summary>
+        SubsystemKind ISubsystem.Kind => SubsystemKind.FeedbackProvider;
+
+        /// <summary>
+        /// Gets feedback based on the given commandline and error record.
+        /// </summary>
+        /// <returns></returns>
+        string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token);
+    }
+
+    internal sealed class GeneralCommandErrorFeedback : IFeedbackProvider
+    {
+        private readonly Guid _guid;
+        private readonly object[] _args;
+        private ScriptBlock? _fuzzySb;
+
+        internal GeneralCommandErrorFeedback()
+        {
+            _guid = new Guid("A3C6B07E-4A89-40C9-8BE6-2A9AAD2786A4");
+            _args = new object[1];
+        }
+
+        public Guid Id => _guid;
+
+        public string Name => "General";
+
+        public string Description => "The built-in general feedback source for command errors.";
+
+        public string? GetFeedback(string commandLine, ErrorRecord lastError, CancellationToken token)
+        {
+            var rsToUse = Runspace.DefaultRunspace;
+            if (rsToUse is null)
+            {
+                return null;
+            }
+
+            if (lastError.FullyQualifiedErrorId == "CommandNotFoundException")
```

Comment:
One of the problems with the previous model is that within a script an error would occur which gets placed on `$Error` and the suggestion would look for that, but the resulting suggestion is confusing to users since the script should be an abstraction layer.  Is this still an issue here for checking last error?

Reply:
I built your branch and it appears this isn't an issue

Reply:
The evaluation won't kick in unless `$?` is false, which indicates the last command run failed.